services/users/password.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package users
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-server/v5/model"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func CheckUserPassword(user *model.User, password string) error {
+	if !model.ComparePassword(user.Password, password) {
+		return InvalidPasswordError
+	}
+
+	return nil
+}
+
+// HashPassword generates a hash using the bcrypt.GenerateFromPassword
+func HashPassword(password string) string {
+	hash, err := bcrypt.GenerateFromPassword([]byte(password), 10)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(hash)
+}
+
+func ComparePassword(hash string, password string) bool {
+	if password == "" || hash == "" {
+		return false
+	}
+
+	err := bcrypt.CompareHashAndPassword([]byte(hash), []byte(password))
```

Comment:
Can we return error from here and handle the `err == nil` logic in the upper layer? Should make for a cleaner API.